core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -66,8 +66,8 @@ final class ClassfileIndexer(file: FileObject) extends SLF4JLogging {
       val interfaceNames: List[ClassName] = interfaces.map(ClassName.fromInternal)(collection.breakOut)
       val superClass = Option(superName).map(ClassName.fromInternal)
 
-      internalRefs.addAll(interfaceNames.asJava)
-      internalRefs.addAll(superClass.toList.asJava)
+      internalRefs.addAll(interfaceNames.map(FullyQualifiedReference(_, None)).asJava)
```

Comment:
This would be only if it is used internally to the classfile indexer.

Since we have to use mutable state when using the Visitor pattern (which is what the ASM library is doing), we may as well use the most efficient mutable collections. Scala's mutable collections have terrible performance.